src/tokenizer.rs

```diff
@@ -580,44 +579,55 @@ macro_rules! advance_regex {
     };
 }
 
-fn advance_comment(code: &str) -> Advancement {
-    if let Some(beginning) = PATTERN_COMMENT_MULTI_LINE_BEGIN.find(code) {
-        if beginning.start() == 0 {
-            let block_count = beginning.end() - beginning.start() - "--[[".len();
-
-            let end_regex = Regex::new(&format!(r"\]={{{}}}\]", block_count)).unwrap();
+#[inline]
+fn parse_single_line_comment(code: &str) -> IResult<&str, &str> {
+    preceded(tag("--"), take_till(|x: char| x.is_ascii_control()))(code)
+}
 
-            let end_find = match end_regex.find(code) {
-                Some(find) => find,
-                None => return Err(TokenizerErrorType::UnclosedComment),
-            };
+#[inline]
+fn parse_multi_line_comment_start(code: &str) -> IResult<&str, &str> {
+    delimited(tag("--["), take_while(|x: char| x == '='), tag("["))(code)
+}
 
-            let comment = &code[beginning.end()..end_find.start()];
+#[inline]
+fn parse_multi_line_comment_body<'a>(
+    code: &'a str,
+    equality_symbol: &'a str,
+) -> IResult<&'a str, &'a str> {
+    recognize(many_till(
+        anychar,
+        recognize(tuple((tag("]"), tag(equality_symbol), tag("]")))),
```

Comment:
I tried to find a way here to return the comment without the ending "]]" and without extra allocations but I couldn't find anything.

Reply:
`.map(|comment| &comment[0..comment.len() - block_count])`?

Reply:
That's basically what I do in the block below, so I guess I'll keep it like this.